src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -798,11 +802,12 @@ private static SyntaxBase JsonContentFunctionVariableGenerator(FunctionCallSynta
             return ConvertJsonToBicepSyntax(functionValue as JToken ?? throw new InvalidOperationException($"Expecting function to return {nameof(JToken)}, but {functionValue?.GetType().ToString() ?? "null"} received."));
         }
 
+        private static readonly JTokenType[] SupportedJsonTokenTypes = new[] { JTokenType.Object, JTokenType.Array, JTokenType.String, JTokenType.Integer, JTokenType.Float, JTokenType.Boolean, JTokenType.Null };
```

Comment:
[nit] Use `ImmutableArray<JTokenType>` - we try to avoid mutable structures wherever possible